crates/runtime/src/component/blobstore.rs

```diff
@@ -77,42 +84,72 @@ impl container::HostContainer for Ctx {
             .table
             .get(&container)
             .context("failed to get container")?;
-        match self.handler.get_data(container, name, start..=end).await {
-            Ok(stream) => {
+        match wrpc::wrpc::blobstore::blobstore::get_container_data(
+            &self.handler,
+            Some(ReplacedInstanceTarget::BlobstoreContainer),
+            &wrpc::wasi::blobstore::types::ObjectId {
+                container: container.to_string(),
+                object: name,
+            },
+            start,
+            end,
+        )
+        .await?
+        {
+            (Ok(stream), io) => {
+                if let Some(io) = io {
+                    // TODO: Move this into the runtime
+                    io.await.context("failed to perform async I/O")?;
+                }
                 let value = self
                     .table
                     .push(stream)
                     .context("failed to push stream and size")?;
                 Ok(Ok(value))
             }
-            Err(err) => Ok(Err(format!("{err:#}"))),
+            (Err(err), _) => Ok(Err(err)),
         }
     }
 
     #[instrument(skip(self))]
     async fn write_data(
         &mut self,
         container: Resource<Container>,
-        name: ObjectName,
+        object: ObjectName,
         data: Resource<types::OutgoingValue>,
     ) -> anyhow::Result<Result<()>> {
-        let mut stream = self
-            .table
-            .get::<AsyncVec>(&data)
-            .context("failed to get outgoing value")?
-            .clone();
-        stream.rewind().await.context("failed to rewind stream")?;
+        // TODO: Stream data
```

Comment:
Let's capture an issue for this one since we will almost definitely need streamed data soon

Reply:
https://github.com/wasmCloud/wasmCloud/issues/2475